src/Fabulous.Avalonia.DataGrid/DataGrid.fs

```diff
@@ -319,6 +319,13 @@ type DataGridModifiers =
     static member inline horizontalGridLinesBrush(this: WidgetBuilder<'msg, IFabDataGrid>, value: IBrush) =
         this.AddScalar(DataGrid.HorizontalGridLinesBrush.WithValue(value))
 
+    /// <summary>Sets the HorizontalGridLinesBrush property.</summary>
+    /// <param name="this">Current widget.</param>
+    /// <param name="value">The HorizontalGridLinesBrush value.</param>
+    [<Extension>]
+    static member inline horizontalGridLinesBrush(this: WidgetBuilder<'msg, IFabDataGrid>, value: Color) =
+        this.AddScalar(DataGrid.HorizontalGridLinesBrush.WithValue(value |> ImmutableSolidColorBrush))
```

Comment:
Instantiating `ImmutableSolidColorBrush` manually here will cause a lot of unnecessary memory usage because this code will be executed on every view evaluation and thus create a new brush each time.

Instead, I would recommend creating another attribute for the same property by using `defineAvaloniaPropertyWidget` (like here https://github.com/fabulous-dev/Fabulous.Avalonia/blob/cd85e8b3f68c6dd54c502d2fa5ebc597875f21c0/src/Fabulous.Avalonia/Views/_TextElement.fs#L13C1-L17), and then introduce 2 extensions:

```fs
static member inline horizontalGridLinesBrush(this: WidgetBuilder<'msg, IFabDataGrid>, content: WidgetBuilder<'msg, #IFabBrush>) =
    this.AddWidget(DataGrid.HorizontalGridLinesBrushWidget.WithValue(content.Compile()))

static member inline horizontalGridLinesBrush(this: WidgetBuilder<'msg, IFabDataGrid>, value: Color) =
    this.horizontalGridLinesBrush(View.ImmutableSolidColorBrush(value))
```

Like this, you would still have the convenience of the extension to be able to use named colors, but the creation of the brush would be delegated to Fabulous.

Same comment applies to all changes in this PR.

Reply:
Agreed. I think we are using this already in most places.

Reply:
Makes sense, thanks. I'll make those modifications and update the PR.

Reply:
@ConradRoesch I noticed that there are some other places we need to update. Happy to take from here and update this PR if that is Ok for you ?

Reply:
Hi @edgarfgp, you are welcome to take it from here. Thanks!

Reply:
This is ready now. had couple name resolution conflicts that now are fixed. Thanks for this PR :)